### ISSUE TITLE
Pipelining for payloads

### ISSUE THREAD

**Issue by mshick** [hapijs/bassmaster#44]
I needed the ability to pipeline the results of one request to the POST / PUT payload of another. I was able to affect this ability with fairly small changes to bassmaster. I have it working well in my production app, and would like to make a PR if this functionality is of interest to the community.

I'm filing this issue first, because writing tests will take additional time, and I would only do that for a PR.

**Comment by mshick** [comment]
I went ahead and worked out a proper PR with tests. Hopefully maintainers will have a chance to review. I've been enjoying the functionality. #45

**Comment by cadecairos** [comment]
It would be nice to use the same syntax in path and payload piplining... of course, that'd be a breaking change and necessitate a v2.0

**Comment by mshick** [comment]
@cadecairos so you would propose using the same syntax (JSONPath) style for the path then? I agree, I think it is more flexible and adhere's to a more established standard.

I can rebase my PR to get in sync with the latest on master. We could release the new functionality as an add on first, and bring the path into line later, or do it all in a 2.0 release.

**Comment by cadecairos** [comment]
I think it'll be better to make it a 2.0 feature, since I don't think it's ideal to have two different syntaxes for  two similar functions.

**Comment by cadecairos** [comment]
It looks like we may just want to go with [Hoek.reach](https://github.com/hapijs/hoek#reachobj-chain-options) instead of jsonpath, which would allow us to do this in a 1.9 without breaking anything.

**Comment by mshick** [comment]
Okay, I'll see what I can do. On first blush jsonpath strikes me as more appropriate for this sort of thing, since it's a fully developed syntax while `reach` is more a utility for digging into objects. But, maybe the full functionality of jsonpath isn't needed here? I think the array handling in reach is new, so I'll have to get familiar with that.

Is the idea then, to maintain `$` as the `root` of results, follow that with an index (starting at 0), and then apply reach to that object? e.g.:

```
var results = [{ foo: { bar: [ 'baz' ] } }, { title: 'wahwah' }];

$0.foo.bar.0 
// 'baz'

$1.title
// 'wahwah'
```

**Comment by mshick** [comment]
Okay, I've rebased off the latest here, and modified my PR to use reach instead of JSONPath. The syntax follows what I posted above. I also added a new test to ensure the reach array refs are working, and everything is still passing at 100%.

Please give it a look. I can add a bit to the docs in a separate PR if this is looking good to you.

**Comment by cadecairos** [comment]
woo! awesome! I'll give it a spin as soon as I have a chance

**Comment by cadecairos** [comment]
Fixed in #45
